x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/SecurityIndexReaderWrapper.java

```diff
@@ -96,7 +96,9 @@ public DirectoryReader apply(final DirectoryReader reader) {
                 }
             }
 
-            return permissions.getFieldPermissions().filter(wrappedReader);
+            var indexVersionCreated = searchExecutionContextProvider.apply(shardId).indexVersionCreated();
```

Comment:
Passing down index version here is a little intrusive. However I don't think it is too bad. In other places of the code base passing down index version is a common pattern. 

The alternative we talked about, which would store coalesced ignored source in a different field, would also be an intrusive change. 

So I'm okay with this approach.